packages/cosmic-proto/src/helpers.ts

```diff
@@ -49,3 +51,51 @@ export const typedJson = <T extends keyof Proto3Shape>(
 export type Base64Any<T> = {
   [Prop in keyof T]: T[Prop] extends Uint8Array ? string : T[Prop];
 };
+
+// TODO make codegen toJSON() return these instead of unknown
+// https://github.com/cosmology-tech/telescope/issues/605
+/**
+ * Mimics behavor of .toJSON(), converting Uint8Array to base64 strings
+ * and bigints to strings
+ */
+export type JsonSafe<T> = {
+  [Prop in keyof T]: T[Prop] extends Uint8Array
+    ? string
+    : T[Prop] extends bigint
+      ? string
+      : T[Prop];
+};
+
+/**
+ * The result of RequestQuery.toJSON(). The type in cosms-types says it returns
+ * `unknown` but it's actually this. The Uint8Array fields are base64 encoded, while
+ * bigint fields are strings.
+ */
+export type RequestQueryJson = JsonSafe<RequestQuery>;
+
+const QUERY_REQ_TYPEURL_RE = /^\/(\w+(?:\.\w+)*)\.Query(\w+)Request$/;
```

Comment:
I don't really get the motivation for this but I'm not opposed.

Consider using [named capture groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) tho.

Reply:
Motivation for the validation inside the function, or the function itself?